security-aot/src/main/java/io/micronaut/security/aot/OpenIdProviderMetadataFetcherCodeGenerator.java

```diff
@@ -80,62 +83,156 @@ public void generate(@NonNull AOTContext context) {
-public void setUriLocalesSupported(@Nullable List<String> uriLocalesSupported);
-public void setClaimsParameterSupported(@Nullable Boolean claimsParameterSupported);
-public void setClaimsSupported(@Nullable List<String> claimsSupported);
-public void setCodeChallengeMethodsSupported(@Nullable List<String> codeChallengeMethodsSupported);
-public void setIntrospectionEndpoint(@Nullable String introspectionEndpoint);
-public void setIntrospectionEndpointAuthMethodsSupported(@Nullable List<String> introspectionEndpointAuthMethodsSupported);
-public void setRevocationEndpoint(@Nullable String revocationEndpoint);
-public void setRevocationEndpointAuthMethodsSupported(@Nullable List<String> revocationEndpointAuthMethodsSupported);
-public void setCheckSessionIframe(@Nullable String checkSessionIframe);
-public void setEndSessionEndpoint(@Nullable String endSessionEndpoint);
-public void setRequestUriParameterSupported(@Nullable Boolean requestUriParameterSupported);
-public void setOpPolicyUri(@Nullable String opPolicyUri);
-public void setOpTosUri(@Nullable String opTosUri);
-public void setRequestParameterSupported(@Nullable Boolean requestParameterSupported);
-public void setRequestObjectEncryptionAlgValuesSupported(@Nullable List<String> requestObjectEncryptionAlgValuesSupported);
-public void setRequestObjectEncryptionEncValuesSupported(@Nullable List<String> requestObjectEncryptionEncValuesSupported);
-public void setRequestObjectSigningAlgValuesSupported(@Nullable List<String> requestObjectSigningAlgValuesSupported);
-                        */
-        return context.javaFile(builder.build());
+                .addMethod(generateMethod(defaultOpenIdProviderMetadata))
+                .build());
     }
 
 
+    private MethodSpec generateMethod(@NonNull DefaultOpenIdProviderMetadata defaultOpenIdProviderMetadata) {
+        MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder("create")
+                .returns(DefaultOpenIdProviderMetadata.class)
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .addStatement("$T metadata = new $T()", DefaultOpenIdProviderMetadata.class, DefaultOpenIdProviderMetadata.class);
+        if (defaultOpenIdProviderMetadata.getUserinfoEndpoint() != null) {
+            methodBuilder.addStatement("metadata.setUserinfoEndpoint($S)", defaultOpenIdProviderMetadata.getUserinfoEndpoint());
+        }
+        if (defaultOpenIdProviderMetadata.getRequireRequestUriRegistration() != null) {
+            methodBuilder.addStatement("metadata.setRequireRequestUriRegistration($S)", defaultOpenIdProviderMetadata.getRequireRequestUriRegistration());
+        }
+        if (defaultOpenIdProviderMetadata.getAuthorizationEndpoint() != null) {
+            methodBuilder.addStatement("metadata.setAuthorizationEndpoint($S)", defaultOpenIdProviderMetadata.getAuthorizationEndpoint());
+        }
+        if (defaultOpenIdProviderMetadata.getUserinfoEncryptionEncValuesSupported() != null) {
```

Comment:
if you moved the null check down to the `addListStringSetterStatement` call, then you could remove a lot of noise in this method.

Reply:
yup